x-pack/plugins/security/public/authentication/login/login_page.test.tsx

```diff
@@ -228,6 +253,73 @@ describe('LoginPage', () => {
     });
   });
 
+  describe('login selector', () => {
+    it('renders as expected with login form', async () => {
+      const coreStartMock = coreMock.createStart();
+      httpMock.get.mockResolvedValue(
+        createLoginState({
+          selector: {
+            enabled: true,
+            providers: [
+              { type: 'saml', name: 'saml1', options: { description: 'Login w/SAML', order: 0 } },
+              { type: 'pki', name: 'pki1', options: { description: 'Login w/PKI', order: 1 } },
+            ],
+          },
+        })
+      );
+
+      const wrapper = shallow(
+        <LoginPage
+          http={httpMock}
+          notifications={coreStartMock.notifications}
+          fatalErrors={coreStartMock.fatalErrors}
+          loginAssistanceMessage=""
+        />
+      );
+
+      await act(async () => {
+        await nextTick();
+        wrapper.update();
+        resetHttpMock(); // so the calls don't show in the BasicLoginForm snapshot
+      });
+
+      expect(wrapper.find(EuiFlexGroup)).toMatchSnapshot();
+    });
+
+    it('renders as expected without login form', async () => {
+      const coreStartMock = coreMock.createStart();
+      httpMock.get.mockResolvedValue(
+        createLoginState({
+          showLoginForm: false,
+          selector: {
+            enabled: true,
+            providers: [
+              { type: 'saml', name: 'saml1', options: { description: 'Login w/SAML', order: 0 } },
+              { type: 'pki', name: 'pki1', options: { description: 'Login w/PKI', order: 1 } },
+            ],
+          },
+        })
+      );
+
+      const wrapper = shallow(
+        <LoginPage
+          http={httpMock}
+          notifications={coreStartMock.notifications}
+          fatalErrors={coreStartMock.fatalErrors}
+          loginAssistanceMessage=""
+        />
+      );
+
+      await act(async () => {
+        await nextTick();
+        wrapper.update();
+        resetHttpMock(); // so the calls don't show in the BasicLoginForm snapshot
+      });
+
+      expect(wrapper.find(EuiFlexGroup)).toMatchSnapshot();
```

Comment:
__note:__ I may need to find a better root element, maybe create a dedicated `LoginSelector` component....